xapian-bindings/java/java.i

```diff
@@ -174,8 +174,11 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next(int) "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
+    Xapian::docid next (int pseudoArg) {
```

Comment:
Ah yes, we already wrap `iterator_valid()` as `hasNext()`.

I do realise that the normal way to use the iterator doesn't exercise this case.  The Java docs seem to specify a behaviour, so the conservative option is to implement what that says, but I don't really know how important it is to do so for a case like this.

The current Java iterator wrappers that dcolish added just return a dummy value from `next()` in this case (e.g. `-1` for `MSetIterator`), so they don't follow that documentation, but they've not been in a stable release yet, so probably aren't widely used.